### PR TITLE
Non Secured APIs Feature

### PR DESCRIPTION
**Pull request by dinusha92**
### Issues

Fixes https://github.com/wso2/product-microgateway/issues/1774

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/AuthFilter.java [hunk 1]
```diff
@@ -82,6 +83,10 @@ private void initializeAuthenticators(APIConfig apiConfig) {
         }
         Authenticator authenticator = new InternalAPIKeyAuthenticator(APIConstants.JwtTokenConstants.INTERNAL_KEY);
         authenticators.add(authenticator);
+    
+        Authenticator nonSecuredAPIAuthenticator = new NonSecuredAPIAuthenticator();
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/AuthFilter.java:87]: Shall we make this class name UnsecuredAPIAuthenticator

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/NonSecuredAPIAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Implements the authenticator interface to authenticate non-secured APIs.
+ */
+
+public class NonSecuredAPIAuthenticator implements Authenticator {
+
+    @Override public boolean canAuthenticate(RequestContext requestContext) {
+        // Retrieve the disable security value. If security is disabled, then you can proceed directly with the
+        // authentication.
+        if (isDisableSecurity(requestContext.getMatchedResourcePath())) {
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/NonSecuredAPIAuthenticator.java:37]: Does this check API level as well

Comment (dinusha92) [reply enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/NonSecuredAPIAuthenticator.java]: From Adapter side, we are setting the resource level values for all the resources. Hence checking only resource level is enough.

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/AuthFilter.java (hunk 1 above)
Comment (dinusha92) [reply enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/AuthFilter.java]: Fixed with https://github.com/wso2/product-microgateway/pull/1784/commits/f109d15dcf8d4806058a4559c61d5cd96408a5c7

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**